stdlib/LinearAlgebra/src/symmetriceigen.jl

```diff
@@ -165,25 +165,55 @@ function eigen!(A::Hermitian{T,S}, B::Hermitian{T,S}; sortby::Union{Function,Not
     GeneralizedEigen(sorteig!(vals, vecs, sortby)...)
 end
 function eigen!(A::RealHermSymComplexHerm{T,<:StridedMatrix}, B::AbstractMatrix{T}; sortby::Union{Function,Nothing}=nothing) where {T<:Number}
-    return _choleigen!(A, B, sortby)
+    return eigen!(Matrix{T}(A), B; sortby) ;
 end
 function eigen!(A::StridedMatrix{T}, B::Union{RealHermSymComplexHerm{T},Diagonal{T}}; sortby::Union{Function,Nothing}=nothing) where {T<:Number}
-    return _choleigen!(A, B, sortby)
+    return eigen!(A, Matrix{T}(B); sortby) ;
 end
-function _choleigen!(A, B, sortby)
-    U = cholesky(B).U
+
+function eigen(A::AbstractMatrix{T}, C::Cholesky{T, <:AbstractMatrix}; sortby::Union{Function,Nothing}=nothing) where {T<:Number}
+    eigen!(copy(A), copy(C); sortby)
```

Comment:
Seems like `_choleigen!` does not modify the Cholesky object, so we don't need to copy it?

```suggestion
    eigen!(copy(A), C; sortby)
```